packages/nimble-components/src/dialog/styles.ts

```diff
@@ -25,14 +27,21 @@ import { themeBehavior } from '../utilities/style/theme';
 export const styles = css`
     ${display('grid')}
 
+    :host {
+        width: ${dialogSmallWidth};
+        height: fit-content;
+        max-height: ${dialogSmallMaxHeight};
+    }
+
     dialog {
         flex-direction: column;
         background-color: ${applicationBackgroundColor};
         border: none;
         box-shadow: ${elevation3BoxShadow};
         padding: 0px;
-        width: 400px;
-        max-height: 600px;
+        width: inherit;
+        height: inherit;
+        max-height: inherit;
```

Comment:
Nothing size related should be relying on inherit. See the discussion in this thread and the related threads it links to:
https://github.com/ni/nimble/pull/1446#discussion_r1309430889

Reply:
In the second link that you provided, I see that the approach was to set the width to 100% and then configure it from the parent component, which would be like a something like a div wrapper. But this approach doesn't seem to work in this case. I've tried setting width, height, max-height of the dialog to 100% and then set them from the host component, but that did not work.

The only other way that I see working is having 2 new tokens, like $ni-nimble-internal-dialog-width, that clients would override, instead of setting the width property on the nimble-dialog. This way, the width & height wouldn't be "inherit", but that token. Does this approach feel suitable to you? If not, could you please provide another suggestion?

Reply:
I'm guessing that setting the width of the `dialog` to `100%` causes it to fill the entire screen width rather than have the same width as the `nimble-dialog`. This is probably because the modal `dialog` has `position: fixed` which means its [containing block is the entire viewport](https://developer.mozilla.org/en-US/docs/Web/CSS/Containing_block#identifying_the_containing_block) rather than its parent DOM element. There's a bit more context [in this SO post](https://stackoverflow.com/questions/73925255/how-can-the-html-dialog-element-be-positioned-according-to-normal-flow-when-op) but I don't like their suggestion to put the `dialog` in a `div` just for positioning.

The reason `inherit` works is because [inherit is always relative to the parent element, not the containing block](https://developer.mozilla.org/en-US/docs/Web/CSS/inherit). But that doesn't solve the risks that Milan mentioned.

I don't love the suggestion to ask clients to set the values of Nimble token. While it's technically possible, we usually try to discourage clients changing the token value; it's easy to mess it up and impact other components.

I haven't been able to think of other ideas so at the moment my vote would be to use `inherit` in the `dialog` but be explicit about setting each of the corresponding attributes on the `:host` (as you're doing), which would mean (I believe) a client would have to actively unset them to cause problems. But it'll be good to see if Milan has other ideas (he was OOO on Friday so let's discuss more next week).

Reply:
I didn't get a chance to explore this more thoroughly today but may be able to tomorrow.

One idea that could be prototyped is having the `nimble-dialog` itself handle it's positioning and override the styles that the `dialog` is doing. The dialog has preset styling but it's nothing special and we partially override it already. I think we could make the `nimble-dialog` the sized and positioned item and then not have to deal with the user trying to size something hidden in the shadow root.

The next not great idea would be to not hide the thing in a shadow root, i.e. expose the internal dialog as a part that could be styled directly. But that seems unnecessary. Seems like the user should just size the `nimble-dialog` and have it be the size visualized.

Reply:
@rajsite any news on the preferred approach? I don't understand how the first idea would be implemented without having the `dialog` inherit the styles of the parent, which is the `nimble-dialog`. Or do you prefer the second approach that you suggested?

Reply:
The first idea works for normal elements. The host element is sized by users and a child element in the shadow root just says make my size width:100%, height:100% to fill the container. Other options are to use display: contents to say layout is delegated to the child.

I prototyped it a bit but ran into issues as the dialog in the shadow root pops out of the context and moves to the top layer. The spec says that if the dialog element is not position fixed the containing block is the [initial containing block](https://drafts.csswg.org/css-position-4/#top-styling) which I believe means the direct parent (nimble-dialog) should control the sizing for something like width:100%, height:100% but that doesn't seem to be the case. 

Regardless, my option 1 suggestion doesn't make much sense because the internal dialog is what moves into the top-layer, not the nimble-dialog wrapper. So we wouldn't want to do all the fixed position styling, etc, on something not in the top layer.


So I think the second approach is what makes sense. We expose the internal dialog as a shadow dom part that can be styled. We should also do something to make the `nimble-dialog` less of a target. Something like set it to visibility:hidden and make it size 0x0, etc. The open questions are the API name choices, ie what to call the shadow part we expose for the dialog. Haven't thought of what to call it but exposing the dialog as a shadow dom part does seem like the probable direction to go.

Reply:
@rajsite Thank you for the detailed explanation. I've updated the implementation to expose the shadow as a part of the nimble-dialog. It can be now styled like this:
![image](https://github.com/ni/nimble/assets/57374665/980194fa-d7a2-464d-97e4-8d34e9052278)

Thus, the need of inheriting the width and height is gone. I hope I understood correctly about the desired approach, let me know if this needs any other modifications.